### PR TITLE
Enhance UI with new hover effects, cursor changes, and other improvements

### LINKED ISSUE #64: Minor UI enhancement: Units filter button
Introduce the following UI changes to increase usability and user experience: 
- [ ] Change filter colour to black upon hover
- [ ] Change cursor to pointer to indicate that it is clickable

Screenshot 2024-06-01 at 9 33 09 PM

Edit: 
- [ ] Also make it unslectable
- [ ] Perhaps maybe also add a hover effect highloght
image

### PR DESCRIPTION
**Pull request by sapta19**
Closes #64 

This pull request enhances the user interface with several improvements to increase usability and user experience. The changes include:

- Change filter color to black upon hover
- Change cursor to pointer to indicate that it is clickable
- Make the filter unselectable
- Add a hover effect highlight to the filter

These changes aim to make the UI more intuitive and visually appealing for users.

**Before:**
- The filter did not change color on hover.
- The cursor did not indicate that the filter was clickable.
- The filter was selectable.
- No hover effect highlight on the filter.
Screenshot 2024-06-02 at 12 54 56 AM

**After:**
- The filter changes to black on hover.
- The cursor changes to a pointer to indicate it is clickable.
- The filter is now unselectable.
- A hover effect highlight has been added to the filter.

Screenshot 2024-06-02 at 12 53 39 AM

### CODE REVIEW

File: app/units/units.css
Comment (jennifermtan) [inline app/units/units.css:127]: Could you remove the background colour as it was not part of the issue? Thanks :)

File: app/units/units.css
Comment (jennifermtan) [inline app/units/units.css:88]: Please remove all comments as it is self-explanatory :)

**Review by jennifermtan — CHANGES_REQUESTED** [review]
Thanks for your changes, and for your suggestion to add user-select: none.

File: app/units/units.css
Comment (jennifermtan) [inline app/units/units.css:125]: We only want the filter icon to be black on hover, however your change also adds the hover effect to the search icon. Please fix this issue.

**Review by jennifermtan — CHANGES_REQUESTED** [review]

File: app/units/units.css
Comment (sapta19) [reply app/units/units.css:127]: Sure I will change it as per your requirement @jennifermtan

File: app/units/units.css
Comment (sapta19) [reply app/units/units.css:88]: Yeah that's right I have added it for your convenience. It will remove it along with the other points that you have mentioned

File: app/units/units.css
Comment (sapta19) [reply app/units/units.css:127]: Screenshot 2024-06-02 at 10 22 01 AM
is this okay for you? @jennifermtan

Comment (jennifermtan) [reply app/units/units.css:127]: Looks great! Maybe decrease the size of the circle (similar to the image I attached previously) and decrease opacity of the colour to increase contrast. Let me know if you have any other ideas as well :)

Comment (sapta19) [reply app/units/units.css:127]: The size of the circle is like the image that you have attached but here as per the theme it is highlighting that the reason it is looking larger.
Not only circle the combination of the circle and the icon should be smaller as like the Screenshot.

Screenshot 2024-06-02 at 11 04 39 AM

is this okay with you ? @jennifermtan

Comment (jennifermtan) [reply app/units/units.css:127]: Nice catch! Let's keep the icon the same size as before but love the highlight :D Thanks so much!

Comment (sapta19) [reply app/units/units.css:127]: I have done a new PR, review the request

Comment (jennifermtan) [reply app/units/units.css:127]: Please do not make a new PR and commit your changes to this branch.

Comment (sapta19) [reply app/units/units.css]: Committed the changes to current brunch

Comment (jennifermtan) [reply app/units/units.css]: Sorry I might've not been too clear but we only want the circular background when the icon is hovered on. It shouldn't be visible when it is not hovered on. Other than that code looks good! I'll approve after you make this last change :) @sapta19

Comment (GoodGameRuler) [reply app/units/units.css]: Looks very good @sapta19. Just this above change.

**Review by GoodGameRuler — DISMISSED** [review]

File: app/units/units.css
Comment (GoodGameRuler) [reply app/units/units.css]: I also suggest that the border-radius should match the other UI buttons. However, I will leave that to @jennifermtan's instruction.

Comment (sapta19) [reply app/units/units.css]: I have changed that @jennifermtan has suggested

Normal State
Screenshot 2024-06-02 at 9 23 14 PM
Hovered State
Screenshot 2024-06-02 at 9 24 15 PM

File: app/units/units.css
Comment (jennifermtan) [inline app/units/units.css:135]: As mentioned in my previous comment, "please use 0.1 opacity for the shadow" :) Everything else looks great - thanks!

**Review by jennifermtan — CHANGES_REQUESTED** [review]

File: app/units/units.css
Comment (sapta19) [reply app/units/units.css:135]: Ok I will make the changes

Comment (sapta19) [reply app/units/units.css]: I have changed the opacity and commit the changes into minor-uiEnhancements branch 
@jennifermtan  You can approve now. let me know if any changes needed.
Screenshot 2024-06-03 at 8 14 40 PM

**Review by jennifermtan — APPROVED** [review]

**Pull request merged**
